In [1]:
import numpy as np
from scipy.stats import norm
import scipy.stats as stats
from statsmodels.stats import weightstats, proportion
from scipy.stats import f

C:\Users\Laercio\AppData\Roaming\Python\Python310\site-packages\pandas\core\arrays\masked.py:61: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


### Questão 1
Uma empresa fabricante de pneus automotivos descobriu que a vida útil dos pneus segue uma distribuição normal com média de 40.000 km e desvio padrão de 3.000 km. Acredita-se que uma mudança no processo de produção resultará em um produto melhor e, por isso, a empresa desenvolveu um novo pneu. Uma amostra de 100 pneus novos foi selecionada. A empresa constatou que a vida útil média desses novos pneus é de 40.900 km. Pode-se concluir que o novo pneu é significativamente melhor que o antigo, usando um nível de significância de 0,01?

H0: μ <= 40.000 km \
HA: μ > 40.000 km

In [2]:
Zcal = (40900 - 40000)/(3000/np.sqrt(100))
Zcal

3.0

In [3]:
Zcrit = norm.ppf(0.99)
Zcrit

2.3263478740408408

Conclusão:
Aqui temos um teste Z de uma cauda (à direita). Como temos Zcal > Zcrit, rejeitamos a hipótese nula. Portanto, o novo pneu é significativamente melhor que o pneu antigo.

### Questão 2
Sabe-se que a alcalinidade, em miligramas por litro, da água nos trechos superiores dos rios de uma determinada região segue uma distribuição normal com um desvio padrão de 10 mg/l. As leituras de alcalinidade nos trechos inferiores dos rios da mesma região também seguem uma distribuição normal, mas com um desvio padrão de 25 mg/l. Dez leituras de alcalinidade são feitas nos trechos superiores de um rio da região e quinze nos trechos inferiores do mesmo rio, com os seguintes resultados.

trechos_sup = [91, 75, 91, 88, 94, 63, 86, 77, 71, 69] \
trechos_inf = [86, 95, 135, 121, 68, 64, 113, 108, 79, 62, 143, 108, 121, 85, 97]

Investigue, com um nível de confiança de 95%, a afirmação de que a verdadeira alcalinidade média da água no curso inferior deste rio é maior do que no curso superior.

H0: μinf <= sup \
HA: μinf > μsup 

In [4]:
trechos_sup = [91, 75, 91, 88, 94, 63, 86, 77, 71, 69] 
trechos_inf = [86, 95, 135, 121, 68, 64, 113, 108, 79, 62, 143, 108, 121, 85, 97]

stats.ttest_ind(trechos_inf, trechos_sup, equal_var=False, alternative='greater')

TtestResult(statistic=2.513931287166985, pvalue=0.010209594136344738, df=20.413605269793052)

Conclusão: \
Temos pValue < 0.05, assim, rejeitamos a hipótese nula. Portanto, a afirmação de que a verdadeira alcalinidade média da água no curso inferior do rio é maior do que no curso superior está correta.

### Questão 3
Abóboras foram cultivadas sob duas condições experimentais, x e y. Do experimento x, uma amostra de 11 abóboras foi coletada e o desvio padrão amostral observado no peso das abóboras foi 0.8. Do experimento y, uma amostra de 9 abóboras foi coletada e o desvio padrão amostral observado foi 0.5. Assumindo que as distribuições de peso são normais, teste a hipótese de que as variâncias verdadeiras são iguais, contra a alternativa de que não são, ao nível de 90% de confiança.

H0: σ1 = σ2 \
Ha: σ1 ≠ σ2

In [5]:
Fcal = 0.8**2/0.5**2
Fcal

2.5600000000000005

In [6]:
Fcrit1 = f.isf(0.05, 10, 8)
Fcrit1

3.347163120233978

In [7]:
Fcrit2 = f.isf(0.95, 10, 8)
Fcrit2

0.3255570361575732

Conclusão: \
Fcal está entre Fcrit1 e Fcrit2. Assim, falhamos em rejeitar a hipótese nula. As variâncias são iguais.

### Questão 4
Queremos testar a afirmação de que ouvir música clássica enquanto estuda faz com que os alunos concluam os trabalhos mais rapidamente. Pedimos a 10 alunos estudem em silêncio no primeiro semestre e estudem ouvindo música clássica no segundo semestre, e registramos o número médio de horas gasto executando trabalhos por semana em cada semestre. Os tempos gastos na execução de trabalhos em 10 semanas de estudo estão listados a seguir. Podemos concluir que estudar ouvindo música clássica reduz o número de horas gastas por semana na execução dos trabalhos escolares? Analise o valor de t_calc (statistic) com confiança de 95%.

Importante: Sua conclusão não deve utilizar o pvalue. Conclua utilizando o valor de t_calc e o t_crítico obtido da tabela t. Note que não é necessário fazer o cálculo à mão, já que o t_calc (statistic) é fornecido pelo Python.

Em silêncio: 14, 13, 16, 21, 15, 19, 11, 20, 19, 16 \
Ouvindo música: 12, 13, 15, 22, 16, 19, 8, 17, 18, 17

H0: μsilencio <= μmusica \
Ha : μsilencio > μmusica

In [8]:
p_silencio = [14, 13, 16, 21, 15, 19, 11, 20, 19, 16]
p_musica = [12, 13, 15, 22, 16, 19, 8, 17, 18, 17]
stats.ttest_rel(p_silencio, p_musica, alternative='greater')

TtestResult(statistic=1.412612867392256, pvalue=0.09570152734951828, df=9)

Conclusão: \
Para alpha = 0,05 com cauda única e df=9, temos t_calc = 1.41 e t_crítico = 1.833. Portanto, falhamos em rejeitar a hipótese nula. Estudar com música não reduz o tempo de execução das tarefas. \
Note que o valor de pValue reafirma essa conclusão.

### Questão 5
O proprietário de um cinema quer saber se há alguma diferença no número de meninos e meninas que assistiram a um novo filme na semana passada. Para tanto, foram amostrados aleatoriamente 76 meninos e 75 meninas e descobriram que 45 meninos e 58 meninas assistiram ao filme. O que o proprietário pode concluir sobre a diferença de proporções com um nível de confiança de 99%?

Confirme o resultado do pvalue confrontando com o resultado do t_calc (statistic) comparado com o t_critico obtido da tabela Z. Só então decida se você falha em rejeitar H0 ou rejeita H0.

H0: p1 - p2 = 0 \
Ha: p1 - p2 ≠ 0

In [9]:
proportion.test_proportions_2indep(45, 76, 58, 75, method='score')

<class 'statsmodels.stats.base.HolderTuple'>
statistic = -2.383248301449112
pvalue = 0.017160615113933176
compare = 'diff'
method = 'score'
variance = 0.005782456140350877
alternative = 'two-sided'
prop1_null = 0.6821192052980133
prop2_null = 0.6821192052980133
tuple = (-2.383248301449112, 0.017160615113933176)
diff = -0.18122807017543863
ratio = 0.7656533575317603
odds_ratio = 0.4254727474972191
value = 0

Conclusão: \
Temos pvalue = 0.017 e alpha = 0.01. A decisão é difícil pois estamos com um empate entre o pvalue e alpha, ainda que pvalue seja um pouquinho maior. Neste caso, vamos analisar o Z_calc (statistic) que é igual a -2.38. Olhando a tabela Z, temos Z_critico = 2.575. Como temos uma análise de duas caudas, falhamos em rejeitar a hipótese nula, portanto não podemos concluir que há diferença na quantidade de meninos e meninas que assistiram o filme.

### Questão 6
Uma turma do quarto ano está estudando o meio ambiente. Uma das tarefas é cultivar feijões em diferentes tipos de solo. Pedro escolheu cultivar seus feijões em terra encontrada do lado de fora da sala de aula, misturada com fiapos de secadora. Ana escolheu cultivar seus feijões em terra para vasos comprada em um viveiro local. Ivo escolheu cultivar seus feijões em terra do jardim de sua mãe. Nenhum produto químico foi usado nas plantas, apenas água. Elas foram cultivadas dentro da sala de aula, perto de uma grande janela. Cada criança cultivou cinco plantas. Ao final do período de crescimento, cada planta foi medida, registrando-se os dados (em polegadas).

pedro = [24, 21, 23, 30, 23] \
ana = [25, 31, 23, 20, 28] \
ivo = [23, 27, 22, 30, 20]

Faça um teste de hipótese capaz de avaliar a se o tipo de solo fez alguma diferença no crescimento do feijão. Utilize confiança de 95%.

H0: μpedro = μana = μivo \
Ha: Médias não são iguais

In [10]:
pedro = [24, 21, 23, 30, 23] 
ana = [25, 31, 23, 20, 28] 
ivo = [23, 27, 22, 30, 20]

stats.f_oneway(pedro, ana, ivo)

F_onewayResult(statistic=0.1339092872570194, pvalue=0.8759582517108224)

Conclusão: \
Temos pValue > 0.05, portanto falhamos em rejeitar a hipótese nula. Não há evidências que o tipo de solo faz diferença no crescimento do feijão.

### Questão 7
Considere o seguinte sistema linear: 

4x - 2y + 3z =  1 \
 x + 3y - 4z = -7 \
3x +  y + 2z =  5 \

Utilizando eliminação gaussiana, obtenha uma matriz triangular, resolva o sistema por substituição e informe a solução do sistema.

Atenção: você deve abrir a célula de Markdown para edição a fim de visualizar a solução.

Sistema:
4x - 2y + 3z = 1
x + 3y - 4z = -7
3x + y + 2z = 5

1. Matriz aumentada:
[4 -2 3 | 1]
[1 3 -4 | -7]
[3 1 2 | 5]

Trocamos L1 com L2 para usar pivô 1:
[1 3 -4 | -7]
[4 -2 3 | 1]
[3 1 2 | 5]

2. Zerar abaixo do pivô (coluna x):
L2 <- L2 - 4L1 = [0 -14 19 | 29]
L3 <- L3 - 3L1 = [0 -8 14 | 26]

3. Zerar abaixo do pivô (coluna y):
L3 <- -14L3 + 8L2 → [0 0 -44 | -132]
Simplificando: [0 0 1 | 3]

4. Retrosubstituição:
z = 3
-14y + 19(3) = 29 → y = 2
x + 3(2) - 4(3) = -7 → x = -1

Solução:
x = -1
y = 2
z = 3